apps/admin-gui/src/app/users/pages/user-detail-page/user-dashboard/dashboard-recently-viewed-button-field/dashboard-recently-viewed-button-field.component.ts

```diff
@@ -64,18 +69,44 @@ export class DashboardRecentlyViewedButtonFieldComponent implements OnInit {
       this.facilitiesIds.push(-1);
     }
 
-    this.vosManager.getVosByIds(this.vosIds).subscribe(vos => {
-      this.existingRecentIds.push(...vos.map(vo => vo.id));
+    this.getVos();
+
+  }
+
+  getVos() {
+    if (this.authResolver.isAuthorized('getVosByIds_List<Integer>_policy', [])) {
+      this.vosManager.getVosByIds(this.vosIds).subscribe(vos => {
+        this.existingRecentIds.push(...vos.map(vo => vo.id));
+        this.getGroups();
+      });
+    } else {
+      this.getGroups();
+    }
+  }
+
+  getGroups() {
+    if (this.authResolver.isAuthorized('getGroupsByIds_List<Integer>_policy', [])) {
       this.groupsManager.getGroupsByIds(this.groupsIds).subscribe(groups => {
         this.existingRecentIds.push(...groups.map(group => group.id));
-        this.facilitiesManager.getFacilitiesByIds(this.facilitiesIds).subscribe(facilities => {
-          this.existingRecentIds.push(...facilities.map(facility => facility.id));
-          recent = recent.filter(rec => this.existingRecentIds.indexOf(rec.id) > -1);
-          this.addRecentlyViewedToDashboard(recent);
-        });
+        this.getFacilities();
       });
-    });
+    } else {
+      this.getFacilities();
+    }
+
+  }
 
+  getFacilities() {
+    if (this.authResolver.isAuthorized('getFacilitiesByIds_List<Integer>_policy', [])) {
+      this.facilitiesManager.getFacilitiesByIds(this.facilitiesIds).subscribe(facilities => {
+        this.existingRecentIds.push(...facilities.map(facility => facility.id));
```

Comment:
This seems weird. The variable `existingRecentIds` contains ids of all objects? Vos, groups, and facilities can share the same ids, which I think might cause trouble.